src/turtle/p3turtle.cc

```diff
@@ -1762,6 +1734,35 @@ void p3turtle::handleTunnelRequest(RsTurtleOpenTunnelItem *item)
 	std::cerr << " new=" << item->partial_tunnel_id << std::dec << std::endl;
 #endif
 
+	// TR forwarding. We must pay attention not to flood the network. The policy is to force a statistical behavior
+	// according to the followin grules:
+	// 	- below a number of tunnel request forwards per second MAX_TR_FORWARD_PER_SEC, we keep the traffic
+	// 	- if we get close to that limit, we drop long tunnels first with a probability that is larger for long tunnels
+	//
+	// Variables involved:
+	// 	distance_to_maximum	: in [0,inf] is the proportion of the current up TR speed with respect to the maximum allowed speed. This is estimated
+	// 				  as an average between the average number of TR over the 60 last seconds and the current TR up speed.
+	// 	corrected_distance 	: in [0,inf] is a squeezed version of distance: small values become very small and large values become very large.
+	// 	depth_peer_probability	: basic probability of forwarding when the speed limit is reached.
+	// 	forward_probability	: final probability of forwarding the packet, per peer.
+	//
+	// When the number of peers increases, the speed limit is reached faster, but the behavior per peer is the same.
+	//
+
+	float forward_probability ;
+
+	{
+		RsStackMutex stack(mTurtleMtx); /********** STACK LOCKED MTX ******/
+		_traffic_info_buffer.tr_dn_Bps += RsTurtleSerialiser().size(item);
```

Comment:
This needs to be kept before exit (same place than before this PR).

Reply:
You are right. I will update the PR.

Reply:
I wrongly introduced authssl.cc fix in the PR. I don't know how to revert that.